pkg/cmd/pulumi/whoami.go

```diff
@@ -58,20 +59,50 @@ func newWhoAmICmd() *cobra.Command {
 			}
 
 			if verbose {
+				if jsonOut {
```

Comment:
This seemed like a good time to add `--json` output to the `pulumi whoami` command. Original behavior without the flag is unaffected.

Reasoning: this means each SDK language implementation can use standard json decoding libraries, instead of manually parsing the current stdout format. I thought this was safest approach. Happy to go the other way if it's deemed best! 👍